src/Task.cpp

```diff
@@ -1474,7 +1474,11 @@ void Task::validate(bool applyDefault /* = true */) {
     setAsNow("end");
 
   // Pending tasks cannot have an end date, remove if present
-  if ((status == Task::pending) && (get("end") != "")) remove("end");
+  if ((status == Task::pending) && (get("end") != "")) {
+    remove("end");
+    throw format("Could not modify task {1}. You cannot set an end date on a pending task.",
+                 this->id);
```

Comment:
To answer your [question from the issue](https://github.com/GothenburgBitFactory/taskwarrior/issues/3584#issuecomment-2347969857), in this context I would prefer a warning since tasks might be `pending` but have an `end` from some other cause, and this will cause Taskwarrior to return an error even when the rnuning `task` command did not modify either of these properties.

That said, as you have seen there are lots of other `throw`s in this function, and I think adding another is the right course. We can think about converting them to warnings later.